internal/internal_task_handlers.go

```diff
@@ -913,7 +944,7 @@ func (w *workflowExecutionContextImpl) ProcessLocalActivityResult(workflowTask *
 		return nil, nil // nothing to do here as we are retrying...
 	}
 
-	err := w.eventHandler.ProcessLocalActivityResult(lar)
+	err := w.getEventHandler().ProcessLocalActivityResult(lar)
```

Comment:
should you do a nil check here?

Reply:
I don't think so. This is called when processing of the decision task, during which the lock of execution context is held and no one else can set the eventHandler to nil. The callback for local activity retry is a special case and a bug we are trying to fix.

Also if eventHandler can be nil here then we are having a serious issue. 

Same thing for rest of the comments.